src/azure.rs

```diff
@@ -201,7 +321,21 @@ impl ObjectStore for MicrosoftAzure {
                     ListState::Done
                 };
 
-                let names = resp.blobs.blobs.into_iter().map(convert_object_meta);
+                let names = resp
+                    .blobs
+                    .blobs
+                    .into_iter()
+                    .map(convert_object_meta)
+                    .filter(|it| {
```

Comment:
I think as written this will filter out errors, which is not ideal. Perhaps something like


```
.filter(|it| match it {
    // This is needed to filter out gen2 directories
    // https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-known-issues#blob-storage-apis
    Ok(meta) => meta.size > 0,
    Err(_) => true,
});
```

Reply:
updated it -  also looks much cleaner now :).

Reply:
updated it -  also looks much cleaner now :).